### ISSUE TITLE
LinAlgError on tracking quickstart, with python 3.4

### ISSUE THREAD

**Issue by arokem** [nipy/dipy#742]
Reported on neurostars: http://neurostars.org/p/3466/#3467

**Comment by arokem** [comment]
I can't replicate this on python 3.4, with numpy 1.10, and scipy 0.16. Let's see what the OP has.

**Comment by arokem** [comment]
No response. I am closing this for now. If anyone sees this, feel free to reopen.

**Comment by MrBago** [comment]
I think this might be related to #852.

**Comment by arokem** [comment]
Of course. Good call.

**Comment by arokem** [comment]
But we'll never know for sure, because the OP never told us what he/she was using :-)

**Comment by pomodorochris** [comment]
I just found this issue because I also experienced #852, which unfortunately has not solved itself yet.
My configuration:
Python 3.4.4 (Anaconda)
scipy 0.17.0
numpy 1.10.4
dipy 0.11.0

OS: ubuntu 14.04 64bit

If I run `tracking_quick_start.py` I get

```
(root)haarburger@PC78:~/Downloads$ ipython tracking_quick_start.py 
/home/students/haarburger/anaconda3/lib/python3.4/site-packages/skimage/filter/__init__.py:6: skimage_deprecation: The `skimage.filter` module has been renamed to `skimage.filters`.  This placeholder module will be removed in v0.13.
  warn(skimage_deprecation('The `skimage.filter` module has been renamed '
Dataset is already in place. If you want to fetch it again please first remove the folder /home/students/haarburger/.dipy/stanford_hardi 
Dataset is already in place. If you want to fetch it again please first remove the folder /home/students/haarburger/.dipy/stanford_hardi 
---------------------------------------------------------------------------
RemoteTraceback                           Traceback (most recent call last)
RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/students/haarburger/anaconda3/lib/python3.4/site-packages/dipy/reconst/csdeconv.py", line 533, in csdeconv
    fodf_sh = _solve_cholesky(P, z)
  File "/home/students/haarburger/anaconda3/lib/python3.4/site-packages/dipy/reconst/csdeconv.py", line 403, in _solve_cholesky
    raise la.LinAlgError(msg)
numpy.linalg.linalg.LinAlgError: 5-th leading minor not positive definite

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/students/haarburger/anaconda3/lib/python3.4/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/students/haarburger/anaconda3/lib/python3.4/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/home/students/haarburger/anaconda3/lib/python3.4/site-packages/dipy/direction/peaks.py", line 323, in _peaks_from_model_parallel_sub
    parallel=False, nbr_processes=None)
  File "/home/students/haarburger/anaconda3/lib/python3.4/site-packages/dipy/direction/peaks.py", line 439, in peaks_from_model
    odf = model.fit(data[idx]).odf(sphere)
  File "/home/students/haarburger/anaconda3/lib/python3.4/site-packages/dipy/reconst/multi_voxel.py", line 18, in new_fit
    return single_voxel_fit(self, data)
  File "/home/students/haarburger/anaconda3/lib/python3.4/site-packages/dipy/reconst/csdeconv.py", line 179, in fit
    P=self._P)
  File "/home/students/haarburger/anaconda3/lib/python3.4/site-packages/dipy/reconst/csdeconv.py", line 536, in csdeconv
    fodf_sh = _solve_cholesky(P, z)
  File "/home/students/haarburger/anaconda3/lib/python3.4/site-packages/dipy/reconst/csdeconv.py", line 403, in _solve_cholesky
    raise la.LinAlgError(msg)
numpy.linalg.linalg.LinAlgError: 9-th leading minor not positive definite
"""

The above exception was the direct cause of the following exception:

LinAlgError                               Traceback (most recent call last)
/home/students/haarburger/Downloads/tracking_quick_start.py in <module>()
     66                              relative_peak_threshold=.5,
     67                              min_separation_angle=25,
---> 68                              parallel=True)
     69 
     70 """

/home/students/haarburger/anaconda3/lib/python3.4/site-packages/dipy/direction/peaks.py in peaks_from_model(model, data, sphere, relative_peak_threshold, min_separation_angle, mask, return_odf, return_sh, gfa_thr, normalize_peaks, sh_order, sh_basis_type, npeaks, B, invB, parallel, nbr_processes)
    408                                           B,
    409                                           invB,
--> 410                                           nbr_processes)
    411 
    412     shape = data.shape[:-1]

/home/students/haarburger/anaconda3/lib/python3.4/site-packages/dipy/direction/peaks.py in _peaks_from_model_parallel(model, data, sphere, relative_peak_threshold, min_separation_angle, mask, return_odf, return_sh, gfa_thr, normalize_peaks, sh_order, sh_basis_type, npeaks, B, invB, nbr_processes)
    216                                repeat(npeaks),
    217                                repeat(B),
--> 218                                repeat(invB)))
    219         pool.close()
    220 

/home/students/haarburger/anaconda3/lib/python3.4/multiprocessing/pool.py in map(self, func, iterable, chunksize)
    258         in a list that is returned.
    259         '''
--> 260         return self._map_async(func, iterable, mapstar, chunksize).get()
    261 
    262     def starmap(self, func, iterable, chunksize=None):

/home/students/haarburger/anaconda3/lib/python3.4/multiprocessing/pool.py in get(self, timeout)
    606             return self._value
    607         else:
--> 608             raise self._value
    609 
    610     def _set(self, i, obj):

LinAlgError: 9-th leading minor not positive definite

```

**Comment by arokem** [comment]
Thanks for the report. Reopened this, and digging around for a good resolution to this. Looks like for your case maybe rebuilding scipy from source?

**Comment by pomodorochris** [comment]
I was hoping to avoid that but it seems to be the only solution (except for switching back to Python 2.7).

**Comment by skoudoro** [comment]
Since we closed #852, I close this one too which seems to be related

**Comment by arokem** [comment]
Hmm. I am not sure it is related... 4th leading minor vs. 9th leading minor (??). 

But I am fine closing this given how long it's been since we've had any activity here.
